src/object.c

```diff
@@ -657,21 +657,11 @@ void dismissSetObject(robj *o, size_t size_hint) {
 }
 
 /* See dismissObject() */
-void dismissZsetObject(robj *o, size_t size_hint) {
+void dismissZsetObject(robj *o) {
     if (o->encoding == OBJ_ENCODING_SKIPLIST) {
         zset *zs = o->ptr;
-        zskiplist *zsl = zs->zsl;
-        serverAssert(zsl->length != 0);
-        /* We iterate all nodes only when average member size is bigger than a
-         * page size, and there's a high chance we'll actually dismiss something. */
-        if (size_hint / zsl->length >= server.page_size) {
-            zskiplistNode *zn = zsl->tail;
-            while (zn != NULL) {
-                dismissSds(zn->ele);
-                zn = zn->backward;
-            }
-        }
-
+        /* In rare cases, the key of the skiplist may exceed the page size(typically 4KB),
+         * so we skip releasing the skiplist and only release the hashtable. */
```

Comment:
This comment explains why we skip releasing the skiplist. The first line was confusing to me because it looks like it focuses on the rare cases where the keys are large. It would clearer to mention the normal case where the keys are small. We can say that it is usually smaller than a page size.

```suggestion
        /* The skiplist nodes are usually smaller than the page size (typically 4KB),
         * so we skip releasing the skiplist and only release the hashtable. */
```